### DISCUSSION TITLE
minijinja still renders templates if values are missing

### DISCUSSION THREAD

**Discussion by Masber** [mitsuhiko/minijinja#632] [Q&A] [answered]
Dear minijinja community,

I am using minijinja in a cli I am working on. 

Currently minijinja will render a template if values are missing by ignoring those values and living them blank in the template, is there a way to tell minijinja to fail if any value is missing?

eg:

based on https://github.com/mitsuhiko/minijinja/tree/main/examples/render-template

I could have :

users.json
```
{
  "users": [
    {"name": "Peter"},
    {"name": "John"}
  ]
}
```

users.html
```
<!doctype html>
<title>User List</title>
<ul>
  {%- for user in users %}
  <li>{{ user.id }}: {{ user.name }}
  {%- endfor %}
</ul>
```

The users.name field is missing in the context/values file, however minijinja still renders the file as:

```
<!doctype html>
<title>User List</title>
<ul>
  <li>: Peter
  <li>: John
</ul>
```

By ignoring the missing `name.id field`.

I would like to know if it is possible to tell minijinja to fail if any variable in the template is not expanded.

thank you

**Comment by mitsuhiko** [comment, marked as answer]
Pass `--strict` to `minijinja-cli` and it will fail in all cases.
